users/views.py

```diff
@@ -42,4 +46,57 @@ def profile(request):
         request,
         'users/profile.html',
         {'piechart_data': json.dumps(piechart_data)},
-        context_instance = RequestContext(request, processors = [custom_proc]))
+        context_instance=RequestContext(request, processors=[custom_proc]))
+
+
+def user_create(request):
+    if request.method == 'POST':
+        user_form = UserCreationForm(request.POST)
+        if user_form.is_valid():
+            user = user_form.save()
+            user.backend = 'django.contrib.auth.backends.ModelBackend'
+            logger.info('user %s created' % str(user))
+            login(request, user)
+            return redirect('/index')
+        else:
+            return render(
+                request, 'users/auth.html',
+                {'form': user_form, 'title': 'Sign Up'},
+                context_instance=RequestContext(request, processors=[custom_proc]))
+    return render(
+        request,
+        'users/auth.html',
+        {'form': UserCreationForm(), 'title': 'Sign Up'},
+        context_instance=RequestContext(request, processors=[custom_proc]))
+
+
+def user_logout(request):
+    logger.info('user %s logged out' % str(request.user))
+    logout(request)
+    return redirect('/index')
+
+
+def user_login(request):
+    if request.user.is_authenticated():
+        return redirect('/index')
+    if request.method == 'POST':
+        user_form = AuthenticationForm(data=request.POST)
+        if user_form.is_valid():
+            user = authenticate(
+                username=user_form.cleaned_data['username'],
+                password=user_form.cleaned_data['password'])
+            user.backend = 'django.contrib.auth.backends.ModelBackend'
+            logger.info('user %s logged in' % str(user))
```

Comment:
I wonder whether there is a manner to record user login IP address, we can keep it in our log.